### PR TITLE
RoboDisplayManager.{start,stop}WifiDisplayScan() should be generated con...

### PR DESCRIPTION
**Pull request by jongerrish**
...ditionally for API 19+

### CODE REVIEW

File: robolectric-shadows/shadows-core/src/main/resources/android/hardware/display/RoboDisplayManager.java.vm [hunk 1]
```diff
@@ -15,7 +15,7 @@ import android.view.Surface;
 
 public class RoboDisplayManager implements IDisplayManager {
 
-#if ($apiLevel >= 21)
+#if ($apiLevel >= 19)
```
Comment (erd) [inline robolectric-shadows/shadows-core/src/main/resources/android/hardware/display/RoboDisplayManager.java.vm]: Could you merge this API > 19 block with the one below it?

File: robolectric-shadows/shadows-core/src/main/resources/android/hardware/display/RoboDisplayManager.java.vm [hunk 2]
```diff
@@ -15,7 +15,7 @@ import android.view.Surface;
...
   // API21+
```
Comment (erd) [inline robolectric-shadows/shadows-core/src/main/resources/android/hardware/display/RoboDisplayManager.java.vm]: This is now a bit misleading.

Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/resources/android/hardware/display/RoboDisplayManager.java.vm]: Ran into some git trouble...

Made the change
Commit
Squash the changes with the original
git push origin fix_broken_robo_display_manager

This change didn't update, but when I went to my own fork, I could see the pull request there, which I made below:-

https://github.com/robolectric/robolectric/pull/1586

I guess you can merge this one, but not sure why it didn't update the original.... thoughts?

**Comment by erd** [comment]
I finally got around to looking at this closer, and I don't think that the change compiles.

[ERROR] /home/travis/build/robolectric/robolectric/robolectric-shadows/shadows-core/target/generated-shadows/android/hardware/display/RoboDisplayManager.java:[10,7] error: RoboDisplayManager is not abstract and does not override abstract method scanWifiDisplays() in IDisplayManager

**Comment by jongerrish** [comment]
Strange.

What version of the Kit Kat android jar are we using? I wonder if when I
was fixing the build scripts I generated the Kit Kat jar based on MR1
release (as this is what the aosp jar we use is named after).

You can also see that this method is defined in MR1 here:-

https://github.com/android/platform_frameworks_base/blob/kitkat-mr1-release/core/java/android/hardware/display/IDisplayManager.aidl

Is it possible that the jar was build from the original Kit Kat release
(but the jar is mis-named) because that one DOESN'T contain this method:-

https://github.com/android/platform_frameworks_base/blob/kitkat-release/core/java/android/hardware/display/IDisplayManager.aidl

Jonathan.

**Comment by erd** [comment]
It is possible that the jar is not built from the MR1 source. The aosp build script documents which tag to build from, but it doesn't attempt to check out the right source. We could rebuild the jar...

**Comment by jongerrish** [comment]
I built the jar locally and everything works fine. I think last time we
spoke you said you were going to rebuild both Kit Kat and Lollipop jars and
publish to Sonatype, there was the bug with the missing classes.

**Comment by erd** [comment]
> I think last time we spoke you said you were going to rebuild both Kit Kat and Lollipop jars

It's on my (ever growing) list...

**Comment by jongerrish** [comment]
Hey Eric,

I built the Kit Kat jar correctly, attached, will save you checking out and
building the code.....could you post this to Sonatype, or otherwise post me
the instructions?

​
 android-all-4.4_r1-robolectric-0.jar
<https://docs.google.com/a/indiekid.org/file/d/0ByMKsN7ljmf3Z250TTM0eFpYRmc/edit?usp=drive_web>
​

**Comment by erd** [comment]
Did you build 4.4_r1 or 4.4.4_r1? I'm looking at the tags in this list: https://source.android.com/source/build-numbers.html#source-code-tags-and-builds.

I've built and uploaded a new version of the Lollipop jar, and am working on the KitKat jar. The jar you provided won't work because it needs to be the signed version of the .bundle file produced by the script. Also, you don't have the key that it is signed with :)

Once both bundles are uploaded, I'll create a PR with the changes to `SdkConfig`.

**Comment by erd** [comment]
I re-built 4.4_r1 this weekend and I'll try to get it uploaded to Sonatype later tonight.

**Comment by erd** [comment]
@jongerrish I tried merging this with the updated 4.4_r1 jars, but it's still complaining that the method in IDisplayManager doesn't exist. Should we update the API 19 jars to 4.4.4_r1?

**Comment by erd** [comment]
If you still want to get this fix merged in, could you find out which version of Android we need to compile? Rebuilding the 4.4_r1 jars didn't have any effect.

**Pull request closed without merging**